buildkite/resource_pipeline_schedule.go

```diff
@@ -249,36 +249,41 @@ func (ps *pipelineSchedule) Delete(ctx context.Context, req resource.DeleteReque
 }
 
 func (ps *pipelineSchedule) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Split import components
+	importComponents := strings.Split(req.ID, "/")
 
-	log.Printf("Import components %s ...", req.ID)
+	// Handle import case with 3 parts (org/pipeline-slug/UUID)
+	if len(importComponents) == 3 && importComponents[0] != "" && importComponents[1] != "" && importComponents[2] != "" {
+		log.Printf("Import components %s ...", req.ID)
 
-	if len(req.ID) == 0 {
-		resp.Diagnostics.AddError(
-			"Missing Import Identifier",
-			fmt.Sprintf("Import identifier is expected with format: $BUILDKITE_ORGANIZATION_SLUG/$BUILDKITE_PIPELINE_SLUG/$PIPELINE_SCHEDULE_UUID. Got: %q", req.ID),
+		apiResponse, err := getPipelineScheduleBySlug(
+			ps.client.genqlient,
+			req.ID,
 		)
-		return
-	}
 
-	apiResponse, err := getPipelineScheduleBySlug(
-		ps.client.genqlient,
-		req.ID,
-	)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Unable to read Pipeline schedule",
+				fmt.Sprintf("Unable to read Pipeline schedule: %s", err.Error()),
+			)
+			return
+		}
 
-	if err != nil {
+		var newReq resource.ImportStateRequest
+		newReq.ID = apiResponse.PipelineSchedule.Id
+		resource.ImportStatePassthroughID(ctx, path.Root("id"), newReq, resp)
+		// Handle import case with 1 component (ID)
+	} else if len(importComponents) == 1 && importComponents[0] != "" {
+		// Pass through ID from import to state for Read
+		resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), importComponents[0])...)
+	} else {
 		resp.Diagnostics.AddError(
-			"Unable to read Pipeline schedule",
-			fmt.Sprintf("Unable to read Pipeline schedule: %s", err.Error()),
+			"Missing Import Identifier",
+			fmt.Sprintf("Import identifier is expected with format: $BUILDKITE_ORGANIZATION_SLUG/$BUILDKITE_PIPELINE_SLUG/$PIPELINE_SCHEDULE_UUID or PIPELINE_SCHEDULE_ID. Got: %q", req.ID),
 		)
 		return
 	}
-
-	var newReq resource.ImportStateRequest
-	newReq.ID = apiResponse.PipelineSchedule.Id
-	resource.ImportStatePassthroughID(ctx, path.Root("id"), newReq, resp)
```

Comment:
FWIW importing a resource is a manual process that I think if we just change straight to graphql ID it would be fine. We don't need to support the slug method as well. For people adopting terraform and importing resources, they are highly likely to check out our docs every time anyway. So I think this could technically be a one-line function of just this line

Reply:
Nice - I guess supporting the older way was more "nice to have" but can whisk this down to just the passthrough 👍

Reply:
Something like the below will make this backwards compatible:

```go
func (ps *pipelineSchedule) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
	if strings.HasPrefix(req.ID, ps.client.organization) {
		importOrg := strings.Split(req.ID, "/")[0]
		if importOrg != ps.client.organization {
			resp.Diagnostics.AddError(
				"Import Organization does not match",
				fmt.Sprintf("Mismatch in import organization. Expected %s, got %s", ps.client.organization, importOrg),
			)
			return
		}
		apiResponse, err := getPipelineScheduleBySlug(
			ps.client.genqlient,
			req.ID,
		)

		if err != nil {
			resp.Diagnostics.AddError(
				"Unable to read Pipeline schedule",
				fmt.Sprintf("Unable to read Pipeline schedule: %s", err.Error()),
			)
			return
		}

		req.ID = apiResponse.PipelineSchedule.Id
		resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
		return
	}
	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
}
```

Reply:
Going for the later - excluding the backwards-compatible implementation (I did it with an import components check but one-liner with the GQL ID is much more simple 👍)